Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/DataExporter.cs

```diff
@@ -65,35 +65,34 @@ public void Export(Stream stream, ExportDataRequest request, Action<ExportProgre
                     {
                         token.ThrowIfCancellationRequested();
 
-                        var objectBatch = pagedDataSource.FetchNextPage();
-
-                        if (objectBatch == null)
+                        while (pagedDataSource.Fetch())
```

Comment:
Probably this line should be used instead of `while (exportedCount < totalCount)`

Reply:
Agreed, Fetch() should replace outer cycle (it is enumerator `bool MoveNext()` analog).